### DISCUSSION TITLE
CompositePolyDataMapper.set_scalars need to be optimized

### DISCUSSION THREAD

**Discussion by Jimmy-KL** [pyvista/pyvista#3789] [Ideas]
I think the method `CompositePolyDataMapper.set_scalars` should be optimized according to the similar method `DataSetMapper.set_scalars`.
For method `CompositePolyDataMapper.set_scalars`, every parameter should be specified while many of them can be optional or give default value. I think it's because the method `CompositePolyDataMapper.set_scalars` is not totally finished.

**Comment by banesullivan** [comment]
I think you're right. Would you be up for taking this on?

**Comment by Jimmy-KL** [comment]
@banesullivan I'd like to, but I don't know how. I'm a geotechnical engineer and still learning coding and how to use Git. I wish in the future I can do some contribution.
